pinot-core/src/main/java/com/linkedin/pinot/core/operator/MCombineOperator.java

```diff
@@ -202,52 +198,29 @@ public IntermediateResultsBlock callJob() throws Exception {
 
       // Get merge results.
       try {
-        _mergedBlock = mergedBlockFuture.get(queryEndTime - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+        mergedBlock = mergedBlockFuture.get(queryEndTime - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
       } catch (InterruptedException e) {
         LOGGER.error("InterruptedException ", e);
-        if (_mergedBlock == null) {
-          _mergedBlock = new IntermediateResultsBlock(e);
-        }
-        List<ProcessingException> exceptions = _mergedBlock.getExceptions();
-        if (exceptions == null) {
-          exceptions = new ArrayList<ProcessingException>();
-        }
-        exceptions.add(QueryException.getException(QueryException.FUTURE_CALL_ERROR, e));
-        _mergedBlock.setExceptionsList(exceptions);
+        mergedBlock = new IntermediateResultsBlock(QueryException.getException(QueryException.FUTURE_CALL_ERROR, e));
```

Comment:
This is the same behavior as before because mergedBlock will always be null. Will revisit this part in the partial result PR.